src/commands/external.rs

```diff
@@ -0,0 +1,56 @@
+use crate::commands::plugins::get_spin_plugins_directory;
+
+use anyhow::{anyhow, Result};
+use spin_plugins::version_check::check_plugin_spin_compatibility;
+use std::{collections::HashMap, env, path::Path};
+use tokio::process::Command;
+use tracing::log;
+
+/// Executes a Spin plugin as a subprocess, expecting the first argument to
+/// indicate the plugin to execute. Passes all subsequent arguments on to the
+/// subprocess.
+pub async fn execute_external_subcommand(args: Vec<String>) -> Result<()> {
```

Comment:
If I mistype a command I now get a message about plugins:

```
$ ./target/debug/spin biuld
Error: The plugin does not exist
```

Could we say something like "unknown command" and print the help, which is what I think we used to do?

Reply:
Yes, will fix this.

Reply:
I am having a hard time replicating the current experience where it will also tell you if you may have mistyped:
```
error: The subcommand 'biuld' wasn't recognized

        Did you mean 'build'?

If you believe you received this message in error, try re-running with 'spin -- biuld'

USAGE:
    spin <SUBCOMMAND>

For more information try --help
```
I wonder if we do want to do `spin plugin run $plugin_name`? This would also resolve the internal plugin name conflict. It might be worth doing for the time being as i am not sure how to dynamically configure clap with the plugin subcommands.

Reply:
I don't know if we need to full-on replicate the current experience, but if we could at least say "unknown command" or "the command '..' wasn't recognised" (and "for more information..."?) that would suffice.  Just trying to avoid sending them in the wrong direction.

Reply:
I added the above, so if its unrecognized, it says so and prints help info. However, i think force mapping plugins would lead to an even better ux

Reply:
I do like the convenience of being able to write `spin test` etc. - from the user point of view it would be nice not to have to care if a command is built in or contributed by a plugin.  I would suggest leaving it as it is, and if the third-party plugin ecosystem takes off, we can add a `spin plugin run` command (and/or `spin plugin alias` or whatever else we come up with after lengthier thought) at that point.  I realise this risks storing up compatibility issues that force-mapping would avoid, though.  Happy to discuss further and more methodically!